src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -113,13 +115,36 @@ public class NuGetPackageManager
             ISolutionManager solutionManager,
             IDeleteOnRestartManager deleteOnRestartManager,
             bool excludeVersion)
+            : this(sourceRepositoryProvider, settings, solutionManager, deleteOnRestartManager, reporter: null, excludeVersion: excludeVersion)
+        {
+        }
+
+        public NuGetPackageManager(
+            ISourceRepositoryProvider sourceRepositoryProvider,
+            ISettings settings,
+            ISolutionManager solutionManager,
+            IDeleteOnRestartManager deleteOnRestartManager,
+            IRestoreProgressReporter reporter) :
+            this(sourceRepositoryProvider, settings, solutionManager, deleteOnRestartManager, reporter, excludeVersion: false)
+        {
+            _ = reporter ?? throw new ArgumentNullException(nameof(reporter));
+        }
+
+        public NuGetPackageManager(
+            ISourceRepositoryProvider sourceRepositoryProvider,
+            ISettings settings,
+            ISolutionManager solutionManager,
+            IDeleteOnRestartManager deleteOnRestartManager,
+            IRestoreProgressReporter reporter,
+            bool excludeVersion)
         {
             SourceRepositoryProvider = sourceRepositoryProvider ?? throw new ArgumentNullException(nameof(sourceRepositoryProvider));
             Settings = settings ?? throw new ArgumentNullException(nameof(settings));
             SolutionManager = solutionManager ?? throw new ArgumentNullException(nameof(solutionManager));
             InstallationCompatibility = PackageManagement.InstallationCompatibility.Instance;
             InitializePackagesFolderInfo(PackagesFolderPathUtility.GetPackagesFolderPath(SolutionManager, Settings), excludeVersion);
             DeleteOnRestartManager = deleteOnRestartManager ?? throw new ArgumentNullException(nameof(deleteOnRestartManager));
+            RestoreProgressReporter = reporter;
```

Comment:
Is it expected that this constructor doesn't throw if `reporter` is null but the other one does? If so, is it worth adding a unit test?

Reply:
Yes, because we don't want to break existing constructors. This is completely optional. 

You're 100% spot on about the tests. I'll add them.